openapi/src/main/java/io/helidon/openapi/ParameterUtil.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.openapi;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Utility methods for dealing with parameters passed to endpoints.
+ */
+public class ParameterUtil {
+
+    /**
+     * Ways to separate multiple values in OpenAPI endpoint parameters.
+     *
+     * Each value has a regex expression to use in separating multiple values
+     * using that technique as well as the lower-case name of the format as
+     * used in OpenAPI parameter definitions.
+     *
+     */
+    private enum Separator {
+        CSV(",", "csv"), // comma-separated values
+        SSV(" ", "ssv"), // space-separated values
+        TSV("\t", "tsv"), // tab-separated values
+        PIPES("\\|", "pipes"), // the escape is needed because we use this in a regex
+        MULTI(",", "multi"); // used only for query params
+
+        private final String separatorString;
+        private final String formatName;
+
+        Separator(String separatorString, String formatName) {
+            this.separatorString = separatorString;
+            this.formatName = formatName;
+        }
+
+
+        /**
+         * Returns the {@code Separator} value that matches the specified lower-
+         * case name.
+         *
+         * @param formatName lower-case name for the separation style
+         * @return {@code Separator} value matching the lower-case name
+         * @throws IllegalArgumentException if no {@code Separator} matches the name
+         */
+        private static Separator match(String formatName) {
```

Comment:
I think you can simplify this whole class by leveraging `Enum.valueOf(Separator.class, formatName.toUpperCase())` as your `match` method. This way you can eliminate the `formatName` field and also support mixed-case names (unless those are prohibited).

Reply:
Yeah, I considered that. The code as written is a bit more general but you're right, given the current choices for format that would be simpler. I'll make that change. If I need to generalize it later we'll cross that bridge then.